File: Golang/GroupAnagrams/GroupAnagrams_test.go
package GroupAnagrams

import (
	"testing"
	"reflect"
)

func TestGroupAnagrams(t *testing.T) {

	if reflect.DeepEqual(groupAnagrams([]string{}),
		[][]string{}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if len(groupAnagrams([]string{"eat", "tea", "tan", "ate", "nat", "bat"})) == 3 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

}

File: Python/GroupAnagramsTest.py
from unittest import TestCase

from GroupAnagrams import GroupAnagrams


class TestGroupAnagrams(TestCase):
    def test_groupAnagrams(self):
        ga = GroupAnagrams()

        self.assertTrue(len(ga.groupAnagrams([])) == 0)

        list0 = ga.groupAnagrams(["eat", "tea", "tan", "ate", "nat", "bat"])
        self.assertTrue(len(list0) == 3)
        self.assertTrue(list0.__contains__(["tan", "nat"]))
        self.assertTrue(list0.__contains__(["eat", "tea", "ate"]))
        self.assertTrue(list0.__contains__(["bat"]))
